sdk/typescript/test/e2e/encrypt-user-share.test.ts

```diff
@@ -89,61 +97,135 @@ describe('Secret key share transfer', () => {
 		const pubKeyRef = await storeEncryptionKey(
 			encryptionKey,
 			EncryptionKeyScheme.Paillier,
-			toolbox.keypair,
-			toolbox.client,
+			dwalletSenderToolbox.keypair,
+			dwalletSenderToolbox.client,
 		);
 		console.log({ pubKeyRef });
 
 		const encryptionKeysHolder = await createActiveEncryptionKeysTable(
-			toolbox.client,
-			toolbox.keypair,
+			dwalletSenderToolbox.client,
+			dwalletSenderToolbox.keypair,
 		);
 
 		await setActiveEncryptionKey(
-			toolbox.client,
-			toolbox.keypair,
+			dwalletSenderToolbox.client,
+			dwalletSenderToolbox.keypair,
 			pubKeyRef?.objectId!,
 			encryptionKeysHolder.objectId,
 		);
 
 		const activeEncryptionKeyAddress = await getActiveEncryptionKeyObjID(
-			toolbox.client,
-			toolbox.keypair.toSuiAddress(),
+			dwalletSenderToolbox.client,
+			dwalletSenderToolbox.keypair.toSuiAddress(),
 			encryptionKeysHolder.objectId,
 		);
 
 		expect(`0x${activeEncryptionKeyAddress}`).toEqual(pubKeyRef?.objectId!);
 	});
 
-	it('encrypts a secret share to a given Sui address successfully', async () => {
-		const [encryptionKey, _] = generate_keypair();
+	it('full flow - encrypts a secret share to a given Sui public key successfully, and store it on chain from the receiving end', async () => {
+		// ======================= Create Source DWallet =======================
+		const [walletCreatorEncryptionKey, walletCreatorDecryptionKey] =
+			generatePaillierKeyPairFromSuiKeyPair(dwalletSenderToolbox.keypair);
+
 		const pubKeyRef = await storeEncryptionKey(
```

Comment:
Create a function that uses existing encryption key if exists, otherwise create and store encryption key. Create an issue on that and point to it with a TODO

Reply:
https://github.com/dwallet-labs/dwallet-network/issues/202